File: A. Police Recruits.c
#include <iostream>
 
using namespace std;
int n,x,c,pos;
int main()
{
    cin>>n;
while(n--){
    cin>>x;
    if (x==-1){
        if(pos>0)--pos;
        else ++c;
    }
    else
        pos +=x;
}
 
cout<<c;
    return 0;
}

File: A. Police Recruits.java
import java.util.Scanner;

public class Main{

	public static void main(String[] args) {
      int n,num,neg=0,pos=0;
        Scanner sc = new Scanner(System.in);
          n=sc.nextInt();
         
         while(n>0)
         {
             num=sc.nextInt();
        	 if(num==-1)//crime accur
        	 {
        		if(pos>0)pos--;
        		else neg++; 
        		 
        	 }
        	 else {
        		 pos+=num;
        	 }
        	 
        	 n--;
         }
        System.out.println(neg);
			sc.close();
	}

}
